src/error.rs

```diff
@@ -5,11 +5,13 @@
 use thiserror::Error;
 
 #[derive(Error, Debug)]
-pub enum IceOryxError {
+pub enum IceoryxError {
     #[error("could not alloce a chunk")]
     ChunkAllocationFailed,
-    #[error("could not create a publisher topic")]
-    PublisherTopicCreationFailed,
+    #[error("could not create a publisher")]
```

Comment:
Since you do not mean any random publisher but a specific one wouldn't "could not create publisher" be more fitting?
The same goes also for the remaining error messages and the chunk one from above.

I think on the high level api we name them samples and not chunks, should we therefore state sample allocation failed?